sklearnex/linear_model/incremental_ridge.py

```diff
@@ -0,0 +1,418 @@
+# ===============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import numbers
+import warnings
+
+import numpy as np
+from sklearn.base import BaseEstimator, MultiOutputMixin, RegressorMixin
+from sklearn.metrics import r2_score
+from sklearn.utils import gen_batches
+from sklearn.utils.validation import check_is_fitted, check_X_y
+
+from daal4py.sklearn._n_jobs_support import control_n_jobs
+from daal4py.sklearn.utils.validation import sklearn_check_version
+
+if sklearn_check_version("1.2"):
+    from sklearn.utils._param_validation import Interval
+
+from onedal.linear_model import IncrementalRidge as onedal_IncrementalRidge
+
+from .._device_offload import dispatch, wrap_output_data
+from .._utils import PatchingConditionsChain
+
+
+@control_n_jobs(
+    decorated_methods=["fit", "partial_fit", "predict", "_onedal_finalize_fit"]
+)
+class IncrementalRidge(MultiOutputMixin, BaseEstimator, RegressorMixin):
```

Comment:
```suggestion
class IncrementalRidge(MultiOutputMixin, RegressorMixin, BaseEstimator):
```

BaseEstimator is generally latest in the MRO for sklearn estimators.  I'm not sure what may overwrite the RegressorMixin or MultiOutputMixin, but just to be sure and to match convention.